pyiron/atomistics/job/sqs.py

```diff
@@ -33,30 +33,79 @@ def pymatgen_to_pyiron(structure):
+    )
     return [pymatgen_to_pyiron(s) for s in structures], decmp, iter_, cycle_time
 
 
 class SQSJob(AtomisticGenericJob):
+    """
+    Produces special quasirandom structures designed to duplicate truly random chemical distributions as well as
+    possible while using finite periodic cells.
+
+    A pyiron wrapper for the [SQS code of Dominik Gehringer](https://git.unileoben.ac.at/p1655622/sqsgenerator).
+
+    'Structural models used in calculations of properties of substitutionally random $A_{1-x}B_x$ alloys are usually
+    constructed by randomly occupying each of the N sites of a periodic cell by A or B. We show that it is possible to
+    design ‘‘special quasirandom structures’’ (SQS’s) that mimic for small N (even N=8) the first few, physically most
+    relevant radial correlation functions of a perfectly random structure far better than the standard technique does.
+    We demonstrate the usefulness of these SQS’s by calculating optical and thermodynamic properties of a number of
+    semiconductor alloys in the local-density formalism.'
+    From the abstract of Zunger, Wei, Ferreira, and Bernard, Phys. Rev. Lett. 65 (1990) 353,
+    DOI:https://doi.org/10.1103/PhysRevLett.65.353
+
+    Input:
+        - mole_fractions (dict): Approximate chemical composition for the output structure(s), using chemical symbols
+            as the keys and floats as the values. Values should sum to 1, but within reason will be automatically
+            adjusted to accommodate the number of atoms in the provided structure (adjustments printed to standard out).
+            Vacancies can also be included using the key '0'.
+        - weights (list/numpy.ndarray): Specifies the weights of the individual shells. (Default is None, which uses the
+            inverse of the shell number for the weight, i.e.
+            [1, 0.5, 0.3333333333, 0.25, 0.2, 0.166666667, 0.1428571429].)
+        - objective (float): Specifies the value the objective functions. The program tries to reach the specified
+            objective function. (Default is 0.)
+        - iterations (int): How many iterations to make searching for the most special quasirandom structure using a
+            random shuffling procedure. (Default is 1e6)
+        - n_output_structures (int): How many different SQS structures to return (in decreasing special
+            quasirandomness). (Default is 1.)
+    """
+
     def __init__(self, project, job_name):
         super(SQSJob, self).__init__(project, job_name)
-        self.input = GenericParameters(table_name="input")
-        self.input['mole_fractions'] = dict()
-        self.input['weights'] = None
-        self.input['objective'] = 0.0 
-        self.input['iterations'] = 1e6 
-        self.input['output_structures'] = 1
+        # self.input = InputList(table_name='input')
+        self.input = InputList(table_name='custom_dict')
+        self.input.mole_fractions = dict()
```

Comment:
How about getting the mole fractions automatically if it remains undefined? Maybe you can include it in `run_static`. And then maybe throw an error if there's only one element.

Reply:
> How about getting the mole fractions automatically if it remains undefined? Maybe you can include it in `run_static`. And then maybe throw an error if there's only one element.

What is the use case for having them automatically determined? I can only picture the following:

```python
ref = pr.create_job(pr.job_type.Vasp, 'ref')
ref.structure = pr.create_ase_bulk('Fe', cubic=True).repeat(3)
ref.structure[:7] = 'Ni'

sqs = pr.create_job(pr.job_type.SQSJob, 'sqs')
sqs.ref_job = ref
sqs.run()
```

Instead of

```python
ref = pr.create_job(pr.job_type.Vasp, 'ref')
ref.structure = pr.create_ase_bulk('Fe', cubic=True).repeat(3)

sqs = pr.create_job(pr.job_type.SQSJob, 'sqs')
sqs.ref_job = ref
sqs.input.mole_fractions = {'Fe': 0.88, 'Ni': 0.12}
sqs.run()
```

Is this enough of an improvement to warrant writing and maintaining the extra interface? I can see in some edge cases that it might be more convenient, but it's also less explicit. Overall I lean towards leaving `mole_fractions` as required input...

Reply:
I might want to have the chemical potential of 2,3,4,.., solute atoms in solid solution for example.

Reply:
> I might want to have the chemical potential of 2,3,4,.., solute atoms in solid solution for example.

I'm not confident I understand -- Do you mean you want run a set of jobs in iteration? In this case I see how it's easy to set up a loop over structures, but I don't think it's so bad to do it with the dicts either. Again, I'm not trying to veto this idea, I'm just still not 100% sure I want to be eternally responsible for maintaining the interface in perpetuity 😝 

e.g.
```python
structures = []
for i in np.arange(5):
    struct[:i] = 'Ni'
    structures.append(struct.copy())
```

vs

```python
fracs = [{'Fe':(1-(i/len(struct)), 'Ni':i/len(struct)} for i in np.arange(5)]
```

Reply:
For example, if I want to calculate the energy of formation for a cluster of atoms, I compare it with that of solid solution. In this case I'd compare the same number of atoms inside the cluster and in the solid solution.

Reply:
Or I can also put it this way: If I already have a structure, I'd naïvely expect the job to figure out what to do, i.e. I'd expect something like this to work:

```
job = pr.create_job('SQSJob', 'sqs')
job.structure = structure
job.run()
```

Reply:
No objection on my end, but I will leave it for another enhancement PR.

Reply:
Fair enough! Any chance you want to implement it? 😉

Reply:
> Fair enough! Any chance you want to implement it? 😉

Sure :)